### ISSUE TITLE
Agent Parsing doesn't allow parameterized keys after v3.44.0

### ISSUE THREAD

**Issue by jkmassel** [buildkite/agent#2556]
We've been running an old version of the agent on some of our infra, and recently discovered a regression in YAML parsing when we upgraded to a newer version of the agent. I haven't binary searched against all releases, but I'm assuming the issue is related to the recent(ish...) YAML parser upgrade?

Consider the following `pipeline.yml` file:

```yaml
# Nodes with values to reuse in the pipeline.
common_params:
  # Common versioned attributes
  - &docker_version "docker#v5.8.0"
  - &ruby_image "public.ecr.aws/docker/library/ruby:3.2.2"

steps:
  - label: "Do the thing"
    command: "whoami"
    plugins:
      - *docker_version :
          image: *ruby_image
```

Running `buildkite-agent pipeline upload --dry-run sample.yaml` after v3.44.0 emits:

```json
{
  "common_params": [
    "docker#v5.8.0",
    "public.ecr.aws/docker/library/ruby:3.2.2"
  ],
  "steps": [
    {
      "command": "whoami",
      "label": "Do the thing",
      "plugins": [
        {
          "github.com/buildkite-plugins/docker_version-buildkite-plugin": {
            "image": "public.ecr.aws/docker/library/ruby:3.2.2"
          }
        }
      ]
    }
  ]
}
```

This is incorrect – the plugin handle should be `docker#v5.8.0`.

The output of `yq -o json` on the same `pipeline.yml` file handles it correctly:

```json
{
  "common_params": [
    "docker#v5.8.0",
    "public.ecr.aws/docker/library/ruby:3.2.2"
  ],
  "steps": [
    {
      "label": "Do the thing",
      "command": "whoami",
      "plugins": [
        {
          "docker#v5.8.0": {
            "image": "public.ecr.aws/docker/library/ruby:3.2.2"
          }
        }
      ]
    }
  ]
}
```

**Comment by mswain** [comment]
Can confirm that I ran into this issue upgrading from 3.38.0 to 3.60.1. To fix, I had to replace all of the YAML aliases used as keys with their anchors manually. I suspect this is the PR that introduced the change:
https://github.com/buildkite/agent/pull/1930

**Comment by jkmassel** [comment]
> Can confirm that I ran into this issue upgrading from 3.38.0 to 3.60.1. To fix, I had to replace all of the YAML aliases used as keys with their anchors manually. I suspect this is the PR that introduced the change: #1930

Thanks @mswain – we ended up having to do the same!! Glad to know we're not the only ones who ran into this issue

**Comment by DrJosh9000** [comment]
Thanks for reporting this and sorry about the breakage! The fix should be in v3.61.0.
